Shared/Services/EthereumService.cs

```diff
@@ -0,0 +1,32 @@
+            try
+            {
+                var contract = web3.Eth.GetContract(contractABI, "0xB25f6D711aEbf954fb0265A3b29F7b9Beba7E55d");
+                var function = contract.GetFunction("uri");
+                object[] parameters = new object[1] { tokenId };
+                var uri = await function.CallAsync<string>(parameters);
+                return uri.Remove(0, 7); //remove the ipfs portion
+            }
+            catch (System.Net.Http.HttpRequestException hex)
+            {
+            }
+            catch (System.Net.Sockets.SocketException sex)
+            {
+            }
+            catch (Exception e)
+            {
+            }
+            return "";
```

Comment:
Wow that raises quite a few eyebrows. 3 different empty exception handlers? And why not make it into Task<string?> so returning null instead of an empty string actually could indicate an error to the outside.